internal/controllers/machineset/machineset_controller.go

```diff
@@ -1232,21 +1226,18 @@ func (r *Reconciler) reconcileV1Beta1Status(ctx context.Context, s *scope) error
 	}
 
 	// Copy the newly calculated status into the machineset
-	if ms.Status.Replicas != newStatus.Replicas ||
+	if ms.Status.Replicas != int32(len(filteredMachines)) ||
```

Comment:
I think we can't hit this code branch anymore?

updateStatus sets ms.Status.Replicas to len(filteredMachines)

Do you think we can just clean all this up to simply modify the status directly?

(nobody cares about that log message on log level 4)

Reply:
Fixed, PTAL